### PR TITLE
create post components

### PR DESCRIPTION
**Pull request by mohmmed23**

### CODE REVIEW

**Review by AseelL — APPROVED** [review]

**Review by AseelL — APPROVED** [review]

**Review by Saeed99Madi — CHANGES_REQUESTED** [review]
As for Interfaces, I think You can build it in one file called `Interfaces.ts` ,instead of create file for every interface.

File: client/src/components/postCard/HoverLikes.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+interface Like{
+  id:number
+  User:User
+}
+interface User{
+  id:number
+  name:string
+  avatar:string
+}
```
Comment (aaamra) [inline client/src/components/postCard/HoverLikes.tsx:11]: you can import here the interfaces you created

don't repeat the interfaces

File: client/src/components/postCard/HoverLikes.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+interface props{
+  likes:Array<Like>
+}
+const HoverLikes = ({ likes }:props) => (
```
Comment (aaamra) [inline client/src/components/postCard/HoverLikes.tsx:15]: if your props are not to many you can inline their type
```suggestion
const HoverLikes = ({ likes }:{ likes: Array<Like> }) => (
```

File: client/src/components/postCard/LikeItem.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  name:string
+  avatar:string
+}
+interface ILike{
+  like: {
+    id:number
+    User:User
+  }
+}
+const LikeItem = ({ like: { id, User } } : ILike) => (
```
Comment (aaamra) [inline client/src/components/postCard/LikeItem.tsx:14]: same issue here like the above one

File: client/src/components/postCard/Post.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+      document.addEventListener('click', handleClick);
+
+      return () => {
+        document.removeEventListener('click', handleClick);
+      };
+    }, []);
+
+    return ref;
+  };
```
Comment (aaamra) [inline client/src/components/postCard/Post.tsx:58]: it's a great idea to create your custom hook

i think we may use this one in other place

so you can create `hooks` folder and create a file on it for this hook

File: client/src/Interfaces/post/Ipost.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    content:string
+    image:string
+    User:User
+    Likes:Array<Like>
+    Tag:Tag
+    Animal:Tag
+  }
+
+}
+export default IPost;
```
Comment (aaamra) [inline client/src/Interfaces/post/Ipost.tsx:17]: rename this file to `IPost.tsx`

**Review by aaamra — CHANGES_REQUESTED** [review]

**Review by AseelL — APPROVED** [review]

**Review by Saeed99Madi — APPROVED** [review]

**Review by aaamra — APPROVED** [review]

**Review by salsabeelomar — CHANGES_REQUESTED** [review]
thanks for your hard work 
you have changes in :
-  You need to separate the Comment component in the folder outside the Post component 
- You can join userInfo for comments and UserPost are the Same logic you need to Create a component for it , and put it outside the Post component

**Review by AseelL — APPROVED** [review]

**Review by AseelL — APPROVED** [review]

**Pull request merged**